### PR TITLE
Permit styles tag 1

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Tag components cleanup
- Adds 'Gas-free approval' tag

## Todo (need help)
- Make this dynamic
`const isPermitCompatible = false // TODO: Make dynamic `

## Preview
Screenshot 2023-08-25 at 15 05 56
Screenshot 2023-08-25 at 15 01 20

### CODE REVIEW

File: apps/cowswap-frontend/src/legacy/components/SearchModal/CurrencyList/styled.ts [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-  padding: 0.25rem 0.3rem 0.25rem 0.3rem;
-  max-width: 6rem;
+  padding: 4px 6px;
   overflow: hidden;
   text-overflow: ellipsis;
   white-space: nowrap;
   justify-self: flex-end;
-  margin-right: 4px;
+  margin: 0 4px 0 0; */
 `
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/components/SearchModal/CurrencyList/styled.ts:15]: Remove?

File: apps/cowswap-frontend/src/legacy/components/SearchModal/CurrencyList/index.tsx [hunk 2]
```diff
@@ -20,29 +20,50 @@ import { TokenAmount } from 'common/pure/TokenAmount'
 import CurrencyListMod, { StyledBalanceText, TagContainer } from './CurrencyListMod'
 import { Tag as TagMod } from './styled'
 
-const UNSUPPORTED_TOKEN_TAG = [
+
+const TOKEN_TAGS: TagInfo[] = [
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/components/SearchModal/CurrencyList/index.tsx:24]: Add an enum for the tag type and transform this into a record

```ts
enum Tags = {
  UNSUPPORTED,
  GAS_FREE,
}

const TOKEN_TAGS: Record<Tags, TagInfo> = {
...
  [Tag.UNSUPPORTED] = {...},
}
```

**Review by alfetopito — APPROVED** [review]
Some nitpicks

File: apps/cowswap-frontend/src/legacy/components/SearchModal/CurrencyList/styled.ts (hunk 1 above)
Comment (fairlighteth) [reply apps/cowswap-frontend/src/legacy/components/SearchModal/CurrencyList/styled.ts:15]: @alfetopito Fixed, please review!

File: apps/cowswap-frontend/src/legacy/components/SearchModal/CurrencyList/index.tsx (hunk 2 above)
Comment (fairlighteth) [reply apps/cowswap-frontend/src/legacy/components/SearchModal/CurrencyList/index.tsx:24]: @alfetopito Thanks for the suggestion, updated please review!

**Review by shoom3301 — APPROVED** [review]

**Review by alfetopito — APPROVED** [review]

File: .gitignore [hunk 3]
```diff
@@ -46,3 +46,4 @@ yalc.lock
 # vercel
 .vercel
 
+analyse.html
```
Comment (alfetopito) [inline .gitignore:49]: This is already on develop https://github.com/cowprotocol/cowswap/blob/7611a3deea3349bc0000ae0296f90209e452e049/.gitignore#L49

Why is it showing up as a change? 🤔

**Pull request merged**